aws/resource_aws_config_remediation_configuration_test.go

```diff
@@ -0,0 +1,324 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/configservice"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func testAccConfigRemediationConfiguration_basic(t *testing.T) {
+	var rc configservice.RemediationConfiguration
+	rInt := acctest.RandInt()
+	expectedName := fmt.Sprintf("tf-acc-test-%d", rInt)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckConfigRemediationConfigurationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccConfigRemediationConfigurationConfig_basic(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckConfigRemediationConfigurationExists("aws_config_remediation_configuration.foo", &rc),
+					testAccCheckConfigRemediationConfigurationName("aws_config_remediation_configuration.foo", expectedName, &rc),
+					resource.TestCheckResourceAttr("aws_config_remediation_configuration.foo", "config_rule_name", expectedName),
+					resource.TestCheckResourceAttr("aws_config_remediation_configuration.foo", "target_id", "SSM_DOCUMENT"),
+					resource.TestCheckResourceAttr("aws_config_remediation_configuration.foo", "target_type", "AWS-PublishSNSNotification"),
+					resource.TestCheckResourceAttr("aws_config_remediation_configuration.foo", "parameters.0.resource_value", "Message"),
+				),
+			},
+		},
+	})
+}
+
+func testAccConfigRemediationConfiguration_importAws(t *testing.T) {
+	resourceName := "aws_config_remediation_configuration.foo"
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckConfigRemediationConfigurationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccConfigRemediationConfigurationConfig_ownerAws(rInt),
+			},
+
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckConfigRemediationConfigurationName(n, desired string, obj *configservice.RemediationConfiguration) resource.TestCheckFunc {
```

Comment:
We typically prefer checking the Terraform state for these values (e.g. with `resource.TestCheckResourceAttr()`) and import testing rather than writing custom functions. Some older resource testing implementations might still be including these custom functions if this was copied from somewhere else.

Reply:
As I'm sure you have noticed, this has been copied then adapted from `resource_aws_config_config_rule` :)
I'll apply the easiest changes first, then the harder ones in a second time (eventually asking for guidance if I don't find an example / documentation by myself)

Reply:
OK I just didn't understand that I checked the attribute in a custom test function AND using the standard way of testing attributes :)